Automattic/wp-calypso
# Checkout: Refactor WebPaymentBox (Apple Pay) as a functional component

As part of an experiment to replace the Apple Pay implementation we currently use (which requires Paygate) with a Stripe solution instead (see #35636), this diff cleans up the rather complex `WebPaymentBox` component to make it easier to follow and to make bugs more apparent.

#### Testing instructions

1. Sandbox the store.
2. Use a proxy to forward requests from `https://wordpress.com` to localhost. This can be a complex step. Let me know if you have any questions about doing this. It's necessary because the button will only appear on a secure connection and on a verified url.
3. Make sure you have a real credit card added to Apple Pay in your local Safari. Look online for how to do this if you don't have it set up already or let me know and I can help.
3. Add a plan to your cart. Make sure you see the "dev" logo in the lower right of the screen.
4. Visit the checkout page in Safari. Apple Pay exists only in Safari on Mac OS.
5. Click the Apple Pay tab above the checkout form.
6. Fill out the form and click the Pay button.
7. When the Apple Pay sheet appears, authorize the payment. Don't worry, you won't be charged as long as you are sandboxing the store.
8. Verify that the transaction goes through correctly and that you are redirected to the next step of the checkout process (which may be a special offer).
